File: ksef_master_backend/Models/ExternalDraft.cs
// Models/ExternalDraft.cs
namespace KSeF.Backend.Models;

public enum ExternalDraftStatus
{
    PENDING,
    APPROVED,
    REJECTED
}

public class ExternalDraft
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string SmartQuoteId { get; set; } = string.Empty;
    public string OfferNumber { get; set; } = string.Empty;
    public ExternalDraftStatus Status { get; set; } = ExternalDraftStatus.PENDING;
    public string IssueDate { get; set; } = string.Empty;
    public string DueDate { get; set; } = string.Empty;

    public string SellerName { get; set; } = string.Empty;
    public string SellerNip { get; set; } = string.Empty;
    public string SellerAddress { get; set; } = string.Empty;
    public string SellerCity { get; set; } = string.Empty;
    public string SellerPostalCode { get; set; } = string.Empty;

    public string BuyerName { get; set; } = string.Empty;
    public string BuyerNip { get; set; } = string.Empty;
    public string BuyerAddress { get; set; } = string.Empty;
    public string BuyerCity { get; set; } = string.Empty;
    public string BuyerPostalCode { get; set; } = string.Empty;

    public List<ExternalDraftItem> Items { get; set; } = new();

    public decimal TotalNet { get; set; }
    public decimal TotalVat { get; set; }
    public decimal TotalGross { get; set; }
    public string Currency { get; set; } = "PLN";
    public int PaymentDays { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ProcessedAt { get; set; }
    public string? ProcessedBy { get; set; }
    public string? RejectionReason { get; set; }
}

public class ExternalDraftItem
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal Quantity { get; set; }
    public string Unit { get; set; } = "szt.";
    public decimal UnitPrice { get; set; }
    public int VatRate { get; set; }
    public decimal Discount { get; set; }
    public decimal TotalNet { get; set; }
    public decimal TotalVat { get; set; }
    public decimal TotalGross { get; set; }
}

File: ksef_master_frontend/src/types/externalDraft.ts
// src/types/externalDraft.ts
export type ExternalDraftStatus = 'PENDING' | 'APPROVED' | 'REJECTED';

export interface ExternalDraftItem {
    name: string;
    description: string | null;
    quantity: number;
    unit: string;
    unitPrice: number;
    vatRate: number;
    discount: number;
    totalNet: number;
    totalVat: number;
    totalGross: number;
}

export interface ExternalDraft {
    id: string;
    smartQuoteId: string;
    offerNumber: string;
    status: ExternalDraftStatus;
    issueDate: string;
    dueDate: string;
    sellerName: string;
    sellerNip: string;
    sellerAddress: string;
    sellerCity: string;
    sellerPostalCode: string;
    buyerName: string;
    buyerNip: string;
    buyerAddress: string;
    buyerCity: string;
    buyerPostalCode: string;
    items: ExternalDraftItem[];
    totalNet: number;
    totalVat: number;
    totalGross: number;
    currency: string;
    paymentDays: number;
    createdAt: string;
    processedAt: string | null;
    processedBy: string | null;
    rejectionReason: string | null;
}

export interface ExternalDraftsResponse {
    success: boolean;
    data?: ExternalDraft[];
    total?: number;
    error?: string;
}

export interface ExternalDraftResponse {
    success: boolean;
    data?: ExternalDraft;
    message?: string;
    error?: string;
}